File: apps/sdkwork-appstore-mini-program/packages/sdkwork-appstore-mp-settings/src/services/settingsService.ts
import type { AppstoreAppClient } from "@sdkwork/appstore-mp-core";

import type { SettingsPageResult } from "../types/settingsModels.js";

/**
 * Settings service.
 *
 * The generated App Store app SDK client is injected by the root bootstrap
 * (`APP_SDK_INTEGRATION_SPEC.md`); this package never constructs a client and
 * never issues raw HTTP.
 */
export interface SettingsService {
  readonly capability: "settings";
}

export function createSettingsService(client: AppstoreAppClient): SettingsService {
  void client;
  return { capability: "settings" };
}

export function emptySettingsResult(): SettingsPageResult<never> {
  return { items: [] };
}

File: apps/sdkwork-appstore-flutter-mobile/packages/sdkwork_appstore_flutter_mobile_settings/lib/src/services/settings_service.dart
import 'package:sdkwork_appstore_flutter_mobile_core/sdkwork_appstore_flutter_mobile_core.dart';

import '../models/settings_models.dart';

/// Settings service.
///
/// The App Store app SDK clients are injected by the root bootstrap
/// (`APP_SDK_INTEGRATION_SPEC.md`); this package never constructs a client and
/// never issues raw HTTP.
class SettingsService {
  const SettingsService({required this.clients});

  final AppstoreAppSdkClients clients;

  String get capability => 'settings';

  SettingsPageResult<Never> empty() =>
      const SettingsPageResult<Never>(items: <Never>[]);
}
